airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -647,6 +646,20 @@ def execute_sync(self, context: Context):
         if self.do_xcom_push:
             return result
 
+    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(max=15), reraise=True)
```

Comment:
We shouldn't retry for every exception, let us add a `retry=tenacity.retry_if_exception_type(kubernetes.client.exceptions.ApiException)`

Reply:
good point thanks! adding it 👍